izanami-frontend/src/pages/users.tsx

```diff
@@ -351,14 +339,34 @@ export function Users() {
                     ),
                   },
                 }}
-                onSubmit={async (ctx) => {
+                onSubmit={(ctx) => {
                   const backendRights = rightStateArrayToBackendMap(ctx.rights);
-                  const payload = {
+                  let payload = {
                     rights: backendRights,
                     admin: ctx.admin,
                     email: ctx.email,
-                    userToCopy: ctx.userToCopy?.value,
                   };
+                  if (ctx.userToCopy) {
+                    readUser
+                      .mutateAsync(ctx.userToCopy.value)
+                      .then((res: TUser) => {
+                        payload = { ...payload, rights: res.rights };
+                        return inviteUserMutation
+                          .mutateAsync(payload)
+                          .then((response) => {
+                            if (response && response.invitationUrl) {
+                              setCreationUrl(response.invitationUrl);
+                            }
+                            setCreating(false);
+                          })
+                          .catch((error) => {
+                            throw new Error(error);
```

Comment:
As I told you in my previous review, this kind of code is not very usefull, since you just rethrow the error.

Reply:
### What do you think about this version : 
```
const inviteUserMutation = useMutation(
    async (data: {
      email: string;
      admin: boolean;
      rights: TRights;
      userToCopy: string;
    }) => {
      if (data.userToCopy) {
        try {
          const res: TUser = await queryUser(data.userToCopy);
          data = { ...data, rights: res.rights };
        } catch (error) {
          throw new Error("Error fetching user rights: " + error);
        }
      }
      return createInvitation(data.email, data.admin, data.rights);
    }
  );
   onSubmit={async (ctx) => {
                  const backendRights = rightStateArrayToBackendMap(ctx.rights);
                  let payload = {
                    rights: backendRights,
                    admin: ctx.admin,
                    email: ctx.email,
                    userToCopy: ctx.userToCopy?.value,
                  };
                  return inviteUserMutation
                    .mutateAsync(payload)
                    .then((response) => {
                      if (response && response.invitationUrl) {
                        setCreationUrl(response.invitationUrl);
                      }
                      setCreating(false);
                    })
                    .catch((error) => {
                      throw new Error(error);
                    });
                }}
```

Reply:
Its better, except that your code is using async/await (see my previous comment about that) and that it remains a catch that just rethrow the error (as I mentionned twice already, I don't see the point).